### PR TITLE
*: Support avg in new plan

### PR DESCRIPTION
**Pull request by shenli**
@coocood @qiuyesuifeng

### CODE REVIEW

File: optimizer/evaluator/evaluator.go [hunk 1]
```diff
@@ -973,3 +975,14 @@ func (e *Evaluator) evalAggSetValue(v *ast.AggregateFuncExpr) {
 	ctx := v.GetContext()
 	v.SetValue(ctx.Value)
 }
+
+func (e *Evaluator) evalAggAvg(v *ast.AggregateFuncExpr) {
+	ctx := v.GetContext()
+	switch x := ctx.Value.(type) {
+	case float64:
+		ctx.Value = x / float64(ctx.Count)
```
Comment (qiuyesuifeng) [inline optimizer/evaluator/evaluator.go]: If `ctx.Count` is zero, can it work well?

File: optimizer/validator.go [hunk 2]
```diff
@@ -32,14 +32,26 @@ type validator struct {
...
 	inPrepare     bool
+	inAggregate   bool
 }
 
 func (v *validator) Enter(in ast.Node) (out ast.Node, skipChildren bool) {
+	switch in.(type) {
+	case *ast.AggregateFuncExpr:
+		if v.inAggregate {
+			// Aggregate function can not contain aggregate function.
+			v.err = ErrInvalidGroupFuncUse
```
Comment (qiuyesuifeng) [inline optimizer/validator.go]: Why not using `invalid use of aggregate function` as error info?

Comment (shenli) [inline optimizer/validator.go]: I copy it from mysql client output.

File: optimizer/evaluator/evaluator.go (hunk 1 above)
Comment (shenli) [inline optimizer/evaluator/evaluator.go]: If Count is zero, ctx.Value must be nil. So it will not be in this switch branch. I will add a test case for it.

**Comment by shenli** [comment]
PTAL @qiuyesuifeng

**Pull request merged**